// clas/build/clas.js
!function(e){if("object"==typeof exports&&"undefined"!=typeof module)module.exports=e();else if("function"==typeof define&&define.amd)define([],e);else{var f;"undefined"!=typeof window?f=window:"undefined"!=typeof global?f=global:"undefined"!=typeof self&&(f=self),f.clas=e()}}(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
function indexOf(arr, item) {
	return Array.prototype.indexOf.call(arr, item);
}

var clas = {
	get : function(element) {
		return element.classList ? element.classList : element.className.split(/\s+/);
	},
	has : function(element, name) {
		return element.classList ? element.classList.contains(name) : !! ~indexOf(clas.get(element), name);
	},
	add : function(element, name) {
		if (element.classList) {
			element.classList.add(name);
			return element;
		}
		var classes = clas.get(element);
		name = name.trim();
		var index = indexOf(classes, name);
		if (!~index) {
			classes.push(name);
			element.className = classes.join(' ');
		}
		return element;
	},
	remove : function(element, name) {
		if (element.classList) {
			element.classList.remove(name);
			return element;
		}
		var classes = clas.get(element);
		name = name.trim();
		var index = indexOf(classes, name);
		if (~index) {
			classes.splice(index, 1);
			element.className = classes.join(' ');
		}
		return element;
	},
	toggle : function(element, name) {
		if (element.classList) {
			element.classList.toggle(name);
			return element;
		}
		name = name.trim();
		return clas.has(element, name) ? clas.remove(element, name) : clas.add(element, name);
	},
};

module.exports = clas;
},{}]},{},[1])(1)
});
